pkgs/top-level/python-packages.nix

```
@@ -12772,6 +12772,8 @@ self: super: with self; {
 
   pyosohotwaterapi = callPackage ../development/python-modules/pyosohotwaterapi { };
 
+  pyotb = callPackage ../development/python-modules/pyotb { inherit python; };
```

```suggestion
  pyotb = callPackage ../development/python-modules/pyotb { };
```
I think that this is not needed. In the derivation, try to use `python3` directly, or maybe `self`.

---

Thanks @GaetanLepage !

Well I not sure I understand because if I don't pass `python` then how would `otb` be built for that specific `python` ?

Because `pyhton3` is just an alias for `python312` right [c.f](https://nixos.org/manual/nixpkgs/stable/#interpreters), if I pass `python3` then `otb`, will it be built for `python313` automatically ?

---

No, if you take `python` as input, it will be substituted with the "current" python.
I.e., when building `python312Packages.pyotb` it will be python 3.12, and when building `python313Packages.pyotb`, it will be python 3.13.

---

> No, if you take python as input, it will be substituted with the "current" python.

Sorry, you mean `python` or `python3` here ?

> I.e., when building python312Packages.pyotb it will be python 3.12, and when building python313Packages.pyotb, it will be python 3.13.

Then isn't this what we want ? Because ideally we want `otb` to be complied with that specific version of python (python312 or python313) and then we build pyotb for that version.

---

> Sorry, you mean `python` or `python3` here ?

I tested with `python`.



> Then isn't this what we want ? Because ideally we want `otb` to be complied with that specific version of python (python312 or python313) and then we build pyotb for that version.

This is what my proposition does, no?
Building `python312Packages.pyotb` will build `otb` with `python` which will be `python312` in this case.
Building `python313Packages.pyotb` will build `otb` with `python` which will be `python313`.

---

> pyotb = callPackage ../development/python-modules/pyotb { };

But you suggested to remove `inherit python` 🤔 

So I'm just confused should I use `python` or `python3` ?

---

Sorry, I was not clear indeed. What I propose is to get rid of this `inherit python` in `python-packages.nix`.
Then, add `python,` to the inputs list in `pyotb/default.nix`.

---

Ah okay. Thanks @GaetanLepage ! I have updated it.